test-data/unit/check-functions.test

```diff
@@ -734,6 +734,188 @@ a = None # type: A
 a.f()
 a.f(None) # E: Too many arguments for "f" of "A"
 
+[case testMethodWithDeclaredDecoratedType]
+
+from typing import Callable, Any
+from mypy_extensions import declared_type
+
+def dec(f): pass
+
+# Note that the decorated type must account for the `self` argument  -- It's applied pre-binding
+class Foo:
+    @declared_type(Callable[[Any, int], str])
```

Comment:
Test similar case where you use a non-`Any` type for the self argument, such as `Foo` here.